inject-groovy/src/main/groovy/io/micronaut/ast/groovy/visitor/GroovyClassElement.java

```diff
@@ -523,6 +529,43 @@ public ClassElement getType() {
                         return visitorContext.getElementFactory().newClassElement(type,
                                 AstAnnotationUtils.getAnnotationMetadata(sourceUnit, compilationUnit, type));
                     }
+
+                    @Override
+                    public Optional<MethodElement> getWriteMethod() {
+                        if (!readOnly) {
+                            return Optional.of(MethodElement.of(
+                                   GroovyClassElement.this,
+                                    annotationMetadata,
+                                    PrimitiveElement.VOID,
+                                    PrimitiveElement.VOID,
+                                    MetaProperty.getSetterName(propertyName),
+                                    ParameterElement.of(getType(), propertyName)
+
+                            ));
+                        }
+                        return Optional.empty();
+                    }
+
+                    @Override
+                    public Optional<MethodElement> getReadMethod() {
+                        return Optional.of(MethodElement.of(
+                                GroovyClassElement.this,
+                                annotationMetadata,
+                                getType(),
+                                getGenericType(),
+                                getGetterName(propertyName, getType())
+                        ));
+                    }
+
+                    private String getGetterName(String propertyName, ClassElement type) {
+                        final String prefix;
```

Comment:
There are methods in NameUtils to do this

Reply:
Changed to use them, hopefully they are consistent with how Groovy behaves